### ISSUE TITLE
Custom projection: `code` should be an alternative to `proj4string`, but either should be acceptable

### ISSUE THREAD

**Issue by prushforth** [Maps4HTML/Web-Map-Custom-Element#377]
I'm trying to write documentation for our implementation of custom projections, so take this not as criticism, just questions that a user might have.  I probably made you make any errors I'm criticizing here so with that, let's begin.

I think the proj4js supports using a defined `code` value instead of providing the `proj4string` definition of the `code`.  I'm not sure what happens if you provide a `code` AND a `proj4string`; probably it uses the `proj4string` and labels the projection by the `code` you give it.  But shouldn't we make it optional to pass in a `code` (perhaps use the required "`projection`" member as the `code` if not provided)?

The `code` string must have a value, but the value can't be the empty string.  I think we should use either the `code` or the `proj4string`, with priority given the the `proj4string`.  Further, I think that "`projection`" should not be of the form "EPSG:nnnn", because I really believe that these should be semantically differentiated from EPSG CRS definitions, so we shouldn't allow the user to make a `projection` name out of an EPSG code, or maybe even not permit a URI (scheme) at all.

Logging the error to the console as you do is quite helpful.

**Comment by prushforth** [comment]
OK I'm [wrong](http://proj4js.org/) about how proj works, I think.  It doesn't appear to have a lot of projections built in, just a few. So the code is required, but maybe in our case we can use the `projection` as the `code` and eliminate the need for the `code` parameter

**Comment by ahmadayubi** [comment]
So you'd have like a map of `projection` to `codes`, hard coded that it would internally pass as the `code` on behalf of the user?

**Comment by ahmadayubi** [comment]
Keep in mind the user can call a projection anything they'd like, not sure how you'd map that to a strict set of codes.

**Comment by prushforth** [comment]
> Keep in mind the user can call a projection anything they'd like

I put some commentary about that in the [documentation](https://maps4html.org/web-map-doc/docs/api/custom-projections), and I omitted the `code` from the docs.  It's necessary to convince the developer/standards community that CRS metadata for the Web must include the `origin`, `resolutions` and `bounds` members, and I would like to lead them away from thinking that an EPSG CRS definition is all that's required for Web mapping. So, I would like to avoid having people use "EPSG:nnnn" as their projection name.  I was thinking that we could banish URI schemes in `projection` names, which I've tried to indicate in the docs.

Something has to change, either the docs or the implementation!
